src/node.cc

```diff
@@ -279,6 +279,8 @@ static v8::Isolate* node_isolate;
 
 DebugOptions debug_options;
 
+static ArrayBufferAllocator array_buffer_allocator;
```

Comment:
This can't be `static`, it's tied to the isolate.  That _will_ be problematic because of the `zeroFill` field - that field is flipped on and off at run-time and that's observable (read: behaves erratically) between isolates.

Reply:
@bnoordhuis  zeroFill is tied with this condition [if (auto zero_fill_field = env->isolate_data()->zero_fill_field())](https://github.com/nodejs/node/blob/master/src/node_buffer.cc#L1100). Currently, the API of CreateIsolateData doesn't support parameter of zero_fill_field for embedding/addon users. It means zeroFill is only tied to the node main thread/environment. This is my understanding at first, and I thought it was not a bad design of node public API.
I can change it to let each isolate have its own ArrayBufferAllocator instance if you think the above  understanding of the current node APIs doesn't work or just not right.

Reply:
@helloshuangzi In that case, it might make sense to expose a way to create a Node.js-style `ArrayBufferAllocator` given an `IsolateData*` pointer?

Reply:
@bnoordhuis , Yes, but 'zeroFill' is only available in node main isolate/environment because of  [if (auto zero_fill_field = env->isolate_data()->zero_fill_field())](https://github.com/nodejs/node/blob/master/src/node_buffer.cc#L1100) based on current node API.
I am thinking that it might be not a bad idea to make all isolates comply with node main isolate/environment about zero_fill_field。

Reply:
The value flipping in the main isolate is still observable from another isolate since `zeroFill[0]` corresponds directly with `&array_buffer_allocator.zero_fill_field_`.  That's why it can't be a static instance.

Reply:
yes, that's what I said all isolates will comply with node main isolate about setting of zero_fill_field, while only node main isolate can change it.

Reply:
> yes, that's what I said all isolates will comply with node main isolate about setting of zero_fill_field, while only node main isolate can change it.

I don’t think that’s a good idea. I agree with Ben, it’s not a good idea for one Isolate to be able to change the allocation mode of other Isolates.

Reply:
To reinforce the point: what you'd see in other isolates is new arraybuffers randomly being filled with garbage instead of zeroes like they're supposed to.

Reply:
Okay, @addaleax and @bnoordhuis, thanks for the feedback about this point. Let me change it to give an ArrayBufferAllocator instance for each isolate. 
@addaleax, as you commented above, we will need:
1. an API to create node.js-style ArrayBufferAllocator, and get its zero_fill_filed.
2. a new API of CreateIsolateData which can take the parameter of zero_fill_field.

Reply:
fixed, please help review!